### PR TITLE
Feature/Enhancement [#285] : Display messages per user breakdown in Pie Chart

### PR DESCRIPTION
**Pull request by Aditya135**
### Description

Added a PieChartWithPaddingAngle in reporting to view messages per user stat

### Issue

[#285 ](https://github.com/papercups-io/papercups/issues/285)

### Screenshots

image
image

## Checklist

- [x] Everything passes when running `mix test`
- [x] Ran `mix format`
- [x] No frontend compilation warnings

### CODE REVIEW

File: assets/src/components/reporting/ReportingDashboard.tsx [hunk 1]
```diff
@@ -133,6 +153,13 @@ class ReportingDashboard extends React.Component<Props, State> {
...
             <MessagesPerDayChart data={this.formatDailyStats()} />
           </Box>
+
+          <Box mb={4} mx={3} sx={{height: 320, flex: 1}}>
+            <Box mb={2}>
+              <Text strong>Messages per user</Text>
+            </Box>
+            <MessagesPerUserChart data={this.formatUserStats()} />
+          </Box>
           {/*
```
Comment (reichert621) [inline assets/src/components/reporting/ReportingDashboard.tsx:163]: let's remove the code on line 188 (https://github.com/papercups-io/papercups/pull/289/files#diff-2160a0e22b7bdfee0ff3c1561656c30fR188) now that this is here :)

File: assets/src/components/reporting/MessagesPerUserChart.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import {useState} from 'react';
+import {PieChart, Pie, Cell, Sector, ResponsiveContainer} from 'recharts';
+import {colors} from '../common';
+import {FAKE_DATA_USERS} from './support';
+
+const COLORS = [colors.green, colors.gold, colors.red, colors.volcano];
+
+// TODO: display messages and conversations in this chart with messages broken
+// down by "sent" vs "received" (where "sent" is outbound, "received" is inbound)
```
Comment (reichert621) [inline assets/src/components/reporting/MessagesPerUserChart.tsx:10]: let's remove this TODO comment now 👍

**Review by reichert621 — COMMENTED** [review]
@Aditya135 looks great!! 🎉 

just a couple minor requests :)

**Comment by Aditya135** [comment]
@reichert621 Done! Can you review the changes? I also added an additional fake data in support to debugging purpose

File: assets/src/components/reporting/MessagesPerUserChart.tsx (hunk 2 above)
Comment (reichert621) [reply assets/src/components/reporting/MessagesPerUserChart.tsx:9]: @Aditya135 can you remove this line as well?

Comment (reichert621) [reply assets/src/components/reporting/MessagesPerUserChart.tsx:9]: actually it's fine, i'll just merge and do it myself 👍

**Comment by reichert621** [comment]
thanks @Aditya135! ❤️

**Pull request merged**